tests/test_util.py

```diff
@@ -659,18 +659,30 @@ def test_get_s3_resource(self):
     def test_get_s3_resource_open_file_error(self):
         """Test get_s3_resource when opening a file fails"""
         self.mock_read_config_value_file.side_effect = OSError('Failed to open file')
-        with self.assertLogs(level=logging.ERROR) as logs:
-            with self.assertRaises(SystemExit):
-                util.get_s3_resource()
-        self.assert_in_element('Unable to load configuration: Failed to open file', logs.output)
+        try:
+            util.get_s3_resource()
+        except util.S3ResourceCreationError as e:
+            # Check if the exception message contains the expected error message
+            if 'Failed to open file' in str(e):
+                # The exception message matches the expected error message
+                pass
+            else:
+                # The exception message does not match the expected error message
+                raise AssertionError(f"Unexpected exception message: {str(e)}")
```

Comment:
It's best to use `self.assertRaises` or `self.assertRaisesRegex` like this:

```suggestion
        expected_err = 'Unable to load configuration: Failed to open file'
        with self.assertRaisesRegex(util.S3ResourceCreationError, expected_err):
            util.get_s3_resource()
```

This way, it ensures an exception is raised as expected, and it ensures it has the expected error message. The current code you have does not actually ensure an exception is raised at all. For example, if you comment out line 661 above, it still passes. When using `self.assertRaises` or `self.assertRaisesRegex`, that is not the case, and it will fail if line 661 is removed.